### PR TITLE
Replace mentions of 'Gradle Enterprise' with 'Develocity'

### PR DESCRIPTION
**Pull request by welandaz**
This PR replaces mentions of Gradle Enterprise with Develocity.

Class names and package names are left as is and should be migrated together with the new version of Develocity Gradle Plugin.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: subprojects/docs/src/docs/stylesheets/userGuideHtmlCommon.xsl [hunk 1]
```diff
@@ -173,7 +173,7 @@
                             </div>
                         </li>
                         <li class="site-header__navigation-item" itemprop="name">
-                            <a target="_top" class="site-header__navigation-link" href="https://gradle.com/enterprise" itemprop="url">Enterprise</a>
+                            <a target="_top" class="site-header__navigation-link" href="https://gradle.com/enterprise" itemprop="url">Develocity</a>
```
Comment (wolfs) [inline subprojects/docs/src/docs/stylesheets/userGuideHtmlCommon.xsl:176]: ❓ Do you need to update the URL as well?

File: subprojects/docs/src/docs/stylesheets/userGuideHtmlCommon.xsl [hunk 2]
```diff
@@ -211,7 +211,7 @@
                         <ul class="site-footer__links-list">
                             <li itemprop="name"><a href="https://gradle.com/build-scans" itemprop="url">Build Scan™</a></li>
                             <li itemprop="name"><a href="https://gradle.com/build-cache" itemprop="url">Build Cache</a></li>
-                            <li itemprop="name"><a href="https://gradle.com/enterprise/resources" itemprop="url">Enterprise Docs</a></li>
+                            <li itemprop="name"><a href="https://gradle.com/enterprise/resources" itemprop="url">Develocity Docs</a></li>
```
Comment (wolfs) [inline subprojects/docs/src/docs/stylesheets/userGuideHtmlCommon.xsl:214]: ❓ Same here, is the URL still valid?

File: subprojects/docs/src/main/resources/footer.html [hunk 3]
```diff
@@ -27,7 +27,7 @@
                 <ul class="site-footer__links-list">
                     <li itemprop="name"><a href="https://gradle.com/build-scans/" itemprop="url">Build Scan™</a></li>
                     <li itemprop="name"><a href="https://gradle.com/build-cache/" itemprop="url">Build Cache</a></li>
-                    <li itemprop="name"><a href="https://gradle.com/enterprise/resources/" itemprop="url">Enterprise Docs</a></li>
+                    <li itemprop="name"><a href="https://gradle.com/enterprise/resources/" itemprop="url">Develocity Docs</a></li>
```
Comment (wolfs) [inline subprojects/docs/src/main/resources/footer.html:30]: ❓ Same question here about the URL.

**Comment by welandaz** [comment]
Closed in favor of opening 2 separate PRs to `release` and `main` with more granular changes.

**Pull request closed without merging**